src/main/java/com/vesta/service/impl/UserServiceImpl.java

```diff
@@ -50,8 +51,12 @@ public UserDto getById(Long id) {
     @Override
     public void create(UserDto userDto) {
         UserEntity entity = userConverter.deconvert(userDto);
-        entity.setPassword(passwordEncoder.encode(userDto.getPassword()));
-        userRepository.save(entity);
+        if (!userRepository.existsByUsername(entity.getUsername())) {
+            entity.setPassword(passwordEncoder.encode(userDto.getPassword()));
+            userRepository.save(entity);
+        } else {
+            throw new VestaException("Username already exists", HttpStatus.UNPROCESSABLE_ENTITY);
```

Comment:
I think better to change UNPROCESSABLE_ENTITY -> CONFLICT or PRECONDITION_FAILED